timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/BlockingTimeLimitedLockService.java

```diff
@@ -227,20 +246,26 @@ private static void logInterrupted(LockRequestSpecification specification, Inter
     }
 
     private BlockingTimeoutException logAndHandleTimeout(LockRequestSpecification specification) {
-        final String logMessage =
-                "Lock service timed out after {} milliseconds" + " when servicing {} for client \"{}\"";
-        log.info(
-                logMessage,
-                SafeArg.of("timeoutDurationMillis", blockingTimeLimitMillis),
-                SafeArg.of("method", specification.method()),
-                SafeArg.of("client", specification.client()),
-                UnsafeArg.of("lockRequest", specification.lockRequest()));
+        final String logMessage = "Lock service timed out after {} milliseconds when servicing {} for client \"{}\"";
+
+        List<Arg<?>> logArgs = new ArrayList<>();
+        logArgs.add(SafeArg.of("timeoutDurationMillis", blockingTimeLimitMillis));
+        logArgs.add(SafeArg.of("method", specification.method()));
+        logArgs.add(SafeArg.of("client", specification.client()));
+
+        if (threadInfoSnapshotManager.isRecordingThreadInfo()) {
+            Set<LockDescriptor> locksToLog = specification.lockRequest().getLockDescriptors().stream()
+                    .limit(MAX_THREADINFO_TO_LOG)
+                    .collect(Collectors.toSet());
+            logArgs.add(threadInfoSnapshotManager.getRestrictedSnapshotAsLogArg(locksToLog));
+        }
+        logArgs.add(UnsafeArg.of("lockRequest", specification.lockRequest()));
+        log.info(logMessage, logArgs);
 
         String errorMessage = String.format(
                 logMessage.replace("{}", "%s"),
-                blockingTimeLimitMillis,
-                specification.method(),
-                specification.client());
+                // Arg.toString() just stringifies the contained value
+                logArgs.toArray());
```

Comment:
This isn't going to add the restricted snapshot, since there aren't enough string format args, so I'd just keep it the way it was.